app/components/footnote_modal_component.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class FootnoteModalComponent < ViewComponent::Base
+  renders_one :body_content
+
+  def initialize(title:)
```

Comment:
Nice to have: allow for some optional HTML classes that will be added to the div with the `modal-dialog` class. The defaults would be `modal-lg modal-dialog-centered` which controls the size and centering of the model. Would allow the size to be tweaked for different parts of the site.